src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -740,12 +855,12 @@ bool Literal::is_fixed_not_numeric() const noexcept {
 }
 
 bool Literal::is_string_like() const noexcept {
-    if (this->null()) {
+    auto const type = this->handle_.node_id().literal_type();
+    if (!type.is_fixed()) {
```

Comment:
`Literal::is_fixed()` has an assertion `!this->null()`. Should we define a deterministic behavior for the `is_*` methods? e.g., they always return false.

Reply:
My general idea was:
- public transformation functions (i.e. `Literal -> Literal functions`) check for null and return back a null literal (for chaining)
- public non-transformation functions (i.e. `Literal -> other type`) usually do not check for null, but if the null-literal has a meaningful value for that function it will behave correctly (e.g. for `null-lit <=> non-null-lit`)
- private functions in general (except where necessary) do not check for null
- i guess public `is_*` functions should check for null to make them harder to misuse

Its probably not entirely consistent right now, so if you find anything more that is inconsistent with these rules feel free to mention it here. Alternatively if you don't agree with them, suggest changes.

- [ ] fix `is_numeric()` to check for null
- [ ] Another thing would be that it is probably a good idea to write those rules down somewhere.